src/resolvers/dag-cbor.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const CID = require('cids')
+const errCode = require('err-code')
+
+const resolve = async (cid, name, path, toResolve, resolve, ipld) => {
+  let node = await ipld.get(cid)
+  let subObject = node
+  let subPath = path
+
+  while (toResolve.length) {
+    const prop = toResolve[0]
+
+    if (prop in subObject) {
```

Comment:
`in` will return `true` for properties in the prototype chain. I think we should only consider own properties.